src/main/java/com/hurko/service/CardServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+package com.hurko.service;
+
+import com.hurko.controller.handler.IncorrectMoneyValue;
+import com.hurko.controller.handler.IncorrectUserException;
+import com.hurko.controller.handler.InvalidInputValueException;
+import com.hurko.model.CardEntity;
+import com.hurko.model.UserEntity;
+import com.hurko.repository.CardRepository;
+import com.hurko.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class CardServiceImpl implements CardService {
+
+    @Autowired
+    private CardRepository cardRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Override
+    public void putMoney(Integer money, String number) {
+        CardEntity card = cardRepository.findByNumber(number);
+        if (Objects.nonNull(card)) {
+            if (money % 100 == 0) {
+                int newBalance = card.getBalance() + money;
+                cardRepository.save(card.setBalance(newBalance));
+            } else {
+                throw new IncorrectMoneyValue("You can only put amount of money which is a multiple of 100");
+            }
+        } else {
+            throw new InvalidInputValueException("Cannot find entity with current set of parameters");
+        }
+    }
+
+    @Override
+    public void getMoney(Principal principal, Integer money, String number) {
+        UserEntity user = userRepository.findByName(principal.getName());
+        CardEntity card = cardRepository.findByNumber(number);
+        if (Objects.nonNull(user) && Objects.nonNull(card)) {
+            if (card.getUser().getId() == user.getId()) {
+                if (card.getBalance() > money && money % 100 == 0) {
+                    int newBalance = card.getBalance() - money;
+                    cardRepository.save(card.setBalance(newBalance));
+                } else {
+                    throw new IncorrectMoneyValue("Please write correct amount of money");
+                }
+            } else {
+                throw new IncorrectUserException("You can't get money from not your card");
+            }
+        } else {
+            throw new InvalidInputValueException("Cannot find entity with current set of parameters");
+        }
```

Comment:
generally this is ok. Can we refactor this somehow to avoid if -> if -> else cascade? This is not mandatory to fix